packages/flutter_tools/lib/runner.dart

```diff
@@ -131,19 +132,23 @@ Future<int> _handleToolError(
 
     // Report to both [Usage] and [CrashReportSender].
     globals.flutterUsage.sendException(error);
-    final CrashReportSender crashReportSender = CrashReportSender(
-      client: http.Client(),
-      usage: globals.flutterUsage,
-      platform: globals.platform,
-      logger: globals.logger,
-      operatingSystemUtils: globals.os,
-    );
-    await crashReportSender.sendReport(
-      error: error,
-      stackTrace: stackTrace,
-      getFlutterVersion: getFlutterVersion,
-      command: args.join(' '),
-    );
+    await asyncGuard(() async {
+      final CrashReportSender crashReportSender = CrashReportSender(
+        client: http.Client(),
+        usage: globals.flutterUsage,
+        platform: globals.platform,
+        logger: globals.logger,
+        operatingSystemUtils: globals.os,
+      );
+      await crashReportSender.sendReport(
+        error: error,
+        stackTrace: stackTrace,
+        getFlutterVersion: getFlutterVersion,
+        command: args.join(' '),
+      );
+    }, onError: (dynamic error) {
+      globals.printError('Error sending crash report: $error');
```

Comment:
Should we print the stacktrace, too?

Reply:
I think its less important than the actual crash - since this is usually an HTTP exception. I want to make sure that users still manually report the real crash.